frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,881 @@
+//!   curator deposit.
+//! - `award_child_bounty` - Close and pay out the specified amount for the completed work.
+//! - `claim_child_bounty` - Claim a specific child-bounty amount from the payout address.
+//! - `unassign_curator` - Unassign an accepted curator from a specific child-bounty.
+//! - `close_child_bounty` - Cancel the child-bounty for a specific treasury amount and close the
+//!   bounty.
+
+// Most of the business logic in this pallet has been
+// originally contributed by "https://github.com/shamb0",
+// as part of the PR - https://github.com/paritytech/substrate/pull/7965.
+// The code has been moved here and then refactored in order to
+// extract child-bounties as a separate pallet.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+mod benchmarking;
+mod tests;
+pub mod weights;
+
+use sp_std::prelude::*;
+
+use frame_support::traits::{
+	Currency, ExistenceRequirement::AllowDeath, Get, OnUnbalanced, ReservableCurrency,
+};
+
+use sp_runtime::{
+	traits::{AccountIdConversion, BadOrigin, Saturating, StaticLookup, Zero},
+	DispatchResult, RuntimeDebug,
+};
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use pallet_bounties::BountyStatus;
+use scale_info::TypeInfo;
+pub use weights::WeightInfo;
+
+pub use pallet::*;
+
+type BalanceOf<T> = pallet_treasury::BalanceOf<T>;
+type BountiesError<T> = pallet_bounties::Error<T>;
+
+/// An index of a bounty. Just a `u32`.
+pub type BountyIndex = u32;
+
+/// A child bounty proposal.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub struct ChildBounty<AccountId, Balance, BlockNumber> {
+	/// The parent of this child-bounty.
+	parent_bounty: BountyIndex,
+	/// The child bounty curator fee.
+	fee: Balance,
+	/// The deposit of child-bounty curator.
+	curator_deposit: Balance,
+	/// The status of this child-bounty.
+	status: ChildBountyStatus<AccountId, BlockNumber>,
+}
+
+/// The status of a child-bounty.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+pub enum ChildBountyStatus<AccountId, BlockNumber> {
+	/// The child-bounty is added and waiting for curator assignment.
+	Added,
+	/// A curator has been proposed by the parent-bounty curator. Waiting for acceptance from the
+	/// child-bounty curator.
+	CuratorProposed {
+		/// The assigned child-bounty curator of this bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is active and waiting to be awarded.
+	Active {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+	},
+	/// The child-bounty is awarded and waiting to released after a delay.
+	PendingPayout {
+		/// The curator of this child-bounty.
+		curator: AccountId,
+		/// The beneficiary of the child-bounty.
+		beneficiary: AccountId,
+		/// When the child-bounty can be claimed.
+		unlock_at: BlockNumber,
+	},
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
```

Comment:
```suggestion
	#[pallet::pallet]
	#[pallet::generate_store(pub(super) trait Store)]
	#[pallet::generate_storage_info]
	pub struct Pallet<T>(_);
```

Without bounded vec stuff, this wont compile. We should make sure this compiles before merge.